internal/pkg/runtime/engines/singularity/process_linux.go

```diff
@@ -425,7 +426,17 @@ func (engine *EngineOperations) PostStartProcess(pid int) error {
 			return err
 		}
 
-		return file.Update()
+		err = file.Update()
+
+		// send SIGUSR1 to the parent process in order to tell it
+		// to detach container process and run as instance.
+		// Sleep a bit in case child would exit
+		time.Sleep(100 * time.Millisecond)
```

Comment:
Okay, looks like you and Cedric decided to move this here and thus `if isInstance && os.Getppid() == ppid` is removed in `master_linux.go`

Reply:
@sasha I have nothing to do with it, I only wrote the unit test.

Reply:
And I do not understand what this comment is about. Is it a question? A suggestion?